### PR TITLE
forecasting operator: bug fix + automlx user specified metric based tuning

### PR DESCRIPTION
**Pull request by codeloop**
forecasting operator: bug fix + automlx user specified metric-based tuning
- Fix for the tuning param in the config spec.
- Fix the neuralforecast missed import.
- metric-based model selection for automlx.
- Ensure local mode of processing for automlx which defaults to Dask based multiprocessing.

### CODE REVIEW

File: ads/opctl/operator/lowcode/forecast/const.py [hunk 1]
```diff
@@ -23,3 +23,13 @@ class SupportedMetrics(str, metaclass=ExtendedEnumMeta):
     RMSE = "rmse"
     MSE = "mse"
     SMAPE = "smape"
+
+
+automlx_metric_dict = {
```
Comment (mrDzurb) [inline ads/opctl/operator/lowcode/forecast/const.py:28]: NIT: we usually use uppercased names for the const variables. 
```
AUTOMLX_METRIC_MAP = {}
```

File: ads/opctl/operator/lowcode/forecast/model/automlx.py [hunk 2]
```diff
@@ -59,7 +61,9 @@ def _build_model(self) -> pd.DataFrame:
                 "" if y_train.index.is_monotonic else "NOT",
                 "monotonic.",
             )
-            model = automl.Pipeline(task="forecasting", n_algos_tuned=n_algos_tuned)
+            model = automl.Pipeline(task="forecasting",
+                                    n_algos_tuned=n_algos_tuned,
+                                    score_metric=automlx_metric_dict[self.spec.metric])
```
Comment (mrDzurb) [inline ads/opctl/operator/lowcode/forecast/model/automlx.py:66]: What if the ```self.spec.metric``` would not presented in the ```automlx_metric_dict```?

File: ads/opctl/operator/lowcode/forecast/model/automlx.py [hunk 3]
```diff
@@ -29,7 +30,8 @@ class AutoMLXOperatorModel(ForecastOperatorBaseModel):
         ),
     )
     def _build_model(self) -> pd.DataFrame:
-        import automl
+        from  automl import init
```
Comment (mrDzurb) [inline ads/opctl/operator/lowcode/forecast/model/automlx.py:33]: Can ```runtime_dependency``` decorator be used here instead?

File: ads/opctl/operator/lowcode/forecast/const.py (hunk 1 above)
Comment (codeloop) [reply ads/opctl/operator/lowcode/forecast/const.py]: Updated to follow the standards.

File: ads/opctl/operator/lowcode/forecast/model/automlx.py (hunk 2 above)
Comment (codeloop) [reply ads/opctl/operator/lowcode/forecast/model/automlx.py]: Updated with dict get to default to `smape`.

File: ads/opctl/operator/lowcode/forecast/model/automlx.py (hunk 3 above)
Comment (codeloop) [reply ads/opctl/operator/lowcode/forecast/model/automlx.py:33]: Runtime dependency decorator is used for `_build_model` @ L24
https://github.com/oracle/accelerated-data-science/blob/4452c73ace5ce3f5621774d935e7cf6ad1bf3c4e/ads/opctl/operator/lowcode/forecast/model/automlx.py#L24

**Review by prasankh — APPROVED** [review]

File: ads/opctl/operator/lowcode/forecast/model/neuralprophet.py [hunk 4]
```diff
@@ -166,7 +167,7 @@ def objective(trial):
                 )
                 study.optimize(
                     objective,
-                    n_trials=self.spec.tuning.n_trials if self.spec.tunning else 10,
+                    n_trials=self.spec.tuning.n_trials if self.spec.tuning else 10,
```
Comment (mrDzurb) [inline ads/opctl/operator/lowcode/forecast/model/neuralprophet.py:170]: NIT: should the ```10``` be moved to the consts?

**Review by mrDzurb — APPROVED** [review]

File: ads/opctl/operator/lowcode/forecast/model/neuralprophet.py (hunk 4 above)
Comment (codeloop) [reply ads/opctl/operator/lowcode/forecast/model/neuralprophet.py]: Updated.

**Pull request merged**